### PR TITLE
Fix PS2Mouse for Real Hardware

### PR DESCRIPTION
**Pull request by JuliusCaesar12**
The mouse stopped working when a scroll wheel wasn't present.

### CODE REVIEW

File: source/Cosmos.HAL2/PS2Mouse.cs [hunk 1]
```diff
@@ -137,19 +144,18 @@ public void HandleMouse(ref INTs.IRQContext context)
             else if (mMouseCycle == 2)
             {
                 mMouseByte[2] = IO.Data.Byte;
-
-                if (HasScrollWheel)
```
Comment (Arawn-Davies) [inline source/Cosmos.HAL2/PS2Mouse.cs:141]: Why remove the check for a ScrollWheel?
Not every PS/2 mouse has one, so please put the checks back in place

**Review by Arawn-Davies — COMMENTED** [review]
Do you have a project I can use to test the changes? I just need something simple and lightweight - the code you used to test this would be ideal 👍🏽

**Review by Arawn-Davies — CHANGES_REQUESTED** [review]
Do you have a project I can use to test the changes? I just need something simple and lightweight - the code you used to test this would be ideal 👍🏽

We do need checks for the mousewheel, as not every PS/2 mouse has one, and there are differences between ps2 and imps2.

**Comment by JuliusCaesar12** [comment]
Re-adding the checks have made the mouse stay at the bottom both in the VM and on my laptop.

**Comment by JuliusCaesar12** [comment]
https://pastebin.com/Y5F709s0
This is a simple test kernel I created to test this (it only uses textmode, so it's not that intensive)

**Comment by quajak** [comment]
Does it mean that we are incorrectly detecting mouse wheels?

**Comment by JuliusCaesar12** [comment]
> Does it mean that we are incorrectly detecting mouse wheels?

I'm not sure. I think it may have something to do with the fact that the trackpad is possibly a Synaptic pad.
I looked at the source code for ReactOS and looked at mouse.c; what I found was that the driver would only switch to finding the z movement if the device was not a generic ps2 device or a ps2pp device.

**Comment by quajak** [comment]
Ok, I suppose we can do something similar. @Arawn-Davies @valentinbreiz @KM198912 thoughts?

**Comment by KM198912** [comment]
I think this could be done and synaptics issues I had before I think there is a push with that

**Comment by KM198912** [comment]
also i think we should make the disabling of that feature a optional user kernel setting since not everyone might have this issue and instead of completly disabling it we should see if we can find a reliable way to detect all buttons, @charlesbetros @quajak thoughts?

**Comment by quajak** [comment]
I think currently the best way would be to set a flag in the mouse driver, which normally is disabled.

**Comment by MishaTY** [comment]
Not needed anymore because of #1989 which allows to disable the scroll wheel.

**Pull request closed without merging**